scrapy/http/request/__init__.py

```diff
@@ -141,3 +170,61 @@ def from_curl(cls, curl_command, ignore_unknown_options=True, **kwargs):
         request_kwargs = curl_to_request_kwargs(curl_command, ignore_unknown_options)
         request_kwargs.update(kwargs)
         return cls(**request_kwargs)
+
+    def to_dict(self, spider: Optional[SpiderType] = None) -> dict:
+        """
+        Return a dictionary containing the Request's data.
+
+        If a spider is given, this method will try to find out the name of the spider method used
+        as callback and include it in the output dict, raising an exception if it cannot be found.
+        """
+        cb = self.callback
+        if callable(cb):
+            cb = _find_method(spider, cb)
+        eb = self.errback
+        if callable(eb):
+            eb = _find_method(spider, eb)
+        d = {
+            "url": to_unicode(self.url),  # urls should be safe (safe_string_url)
+            "callback": cb,
+            "errback": eb,
+            "headers": dict(self.headers),
+            "_encoding": self._encoding,
```

Comment:
I'm not sure `_encoding` is handled properly in the new from_dict implementation

Reply:
Good catch. Is working correctly because `encoding` is in `Request._attributes`, and so it gets set in the next `for` loop, but this `_encoding` assignment is unnecessary. Thanks for noticing.